### DISCUSSION TITLE
Flag to disable default venv detection

### DISCUSSION THREAD

**Discussion by uranusjr** [brettcannon/python-launcher#261] [Ideas]
I’ve had a few times when I ran `py` in a random terminal to test something, and only realised that does not invoke my latest Python version because the shell is in a random project directory. It’d be nice if there’s a switch I can pass to make the behaviour consistent regardless of what cwd I’m in.

**Comment by brettcannon** [comment]
Actually, if you specify any Python version, e.g. `py -3`, it will bypass `.venv` and only search `$PATH`.

```console
/tmp via 🐍 v3.11.1
❯ py --version
Python 3.11.1

/tmp via 🐍 v3.11.1
❯ py --list
 3.11 │ /home/linuxbrew/.linuxbrew/bin/python3.11
 3.10 │ /home/linuxbrew/.linuxbrew/bin/python3.10
 3.9  │ /home/linuxbrew/.linuxbrew/bin/python3.9
 3.8  │ /usr/bin/python3.8
 3.7  │ /usr/bin/python3.7
 3.6  │ /usr/bin/python3.6

/tmp via 🐍 v3.11.1
❯ py -3.10 -m venv .venv

/tmp via 🐍 v3.10.9
❯ py --version
Python 3.10.9

/tmp via 🐍 v3.10.9
❯ py -3 --version
Python 3.11.1
```

Would you still want an explicit flag, or would an FAQ entry about this work for you?

**Reply by uranusjr** [reply]
A FAQ entry is probably a good idea (although I already know that personally). Adding `-3` is probably good enough since we’re unlikely to have Python 4 anytime soon, but I still feel it’s more like a hack than a solution. (Also I might question why ading a version specifier disables venv detection since the Python in the venv is still Python 3…?)

**Reply by brettcannon** [reply]
> A FAQ entry is probably a good idea

👍 https://github.com/brettcannon/python-launcher/issues/264

> I still feel it’s more like a hack than a solution

Oh, it definitely is. 😁 I just have to think about whether this warrants something explicit like an environment variable to switch off (which is probably how I would solve it since I really don't want to clutter the CLI UX and make it harder to reason about what may or may not be passed down to the interpreter).

And to add more personal motivation, this could also be easily solved by https://github.com/brettcannon/python-launcher/discussions/222 if you explicitly only want interpreters and not environments.

> I might question why ading a version specifier disables venv detection since the Python in the venv is still Python 3…?

My thinking (and personal habit) is if I care about a specific version of Python then I'm already signalling that I don't care about my local virtual environment, else I would let the Launcher "fall through" to that virtual environment by letting the Launcher do its thing and select it.

**Comment by brettcannon** [comment]
After [running a poll](*https://fosstodon.org/@brettcannon/110687441217882695), what I think I may do is make the default semantics use the virtual environment if it meets the version restriction, and then provide a `PY_SKIP_VENV` flag to completely disable using `.venv`.
